sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusMessageBatch.cs

```diff
@@ -105,8 +128,15 @@ public bool TryAddMessage(ServiceBusMessage message)
             {
                 AssertNotLocked();
 
-                _scopeFactory.InstrumentMessage(message);
-                return _innerBatch.TryAddMessage(message);
+                var identifier = EntityScopeFactory.InstrumentMessage(message, _entityPath, _fullyQualifiedNamespace);
```

Comment:
Same question here about why we need to do the business with storing off the diagnostic identifiers.

Reply:
I understand now - we need to output the identifiers so that we can link them to the send activity. We used to rely on the SBMessages being cached in the batch, but this is no longer available. 

We have two options:
1) Do what you have here where we store off the identifiers
2) Update to pull the identifiers off of the AmqpMessage.UserProperties.

I'm fine with either approach.